connection_generator.py

```diff
@@ -0,0 +1,197 @@
+"""
+
+from itertools import islice
+import random
+
+
+class ConnectionGenerator:
+    """
+    A class to generate random connections between node ids, based on distribution maps.
+
+    ...
+
+    Attributes:
+        dataset_count: Integer of how many datasets are in a graph.
+        dataset_count_map: Dictionary that maps number of datasets in collection to count of dataset collections.
+        system_count: Integer of how many systems are in a graph.
+        system_count_map: Dictionary that maps number of systems in collection to count of system collections.
+        dataset_read_count: Integer of how many dataset reads are in a graph.
+        dataset_write_count: Integer of how many dataset writes are in a graph.
+        system_input_count: Integer of how many system inputs are in a graph.
+        system_output_count: Integer of how many system outputs are in a graph.
+        dataset_read_count_map: Dictionary that maps number of system inputs of dataset read to count of dataset reads.
+        system_input_count_map: Dictionary that maps number of dataset reads by system input to count of system inputs.
+        dataset_write_count_map: Dictionary that maps number of system outputs of dataset write to count of dataset writes.
+        system_output_count_map: Dictionary that maps number of dataset writes by system output to count of system outputs.
+        datasets_conn_collection: Dictionary that maps dataset collection id to dataset ids.
+        systems_conn_collection: Dictionary that maps system collection id to system ids.
+        dataset_read_conn_systems: Dictionary that maps dataset read id to system ids this dataset inputs to.
+        dataset_write_conn_systems: Dictionary that maps dataset write id to system ids this dataset is output from.
+
+    Methods:
+        get_one_to_many_connections()
+            Creates connections between an element and a group. Each element belongs to one group exactly.
+
+        get_many_to_many_connections()
+            Creates connections between two groups with many to many relationship.
+
+        _dataset_to_dataset_collection()
+            Generates dataset - dataset collection connections.
+
+        _system_to_system_collection()
+            Generates system - system collection connections.
+
+        _dataset_read_to_system_input()
+            Generates connections between dataset reads and system inputs.
+
+        _dataset_write_to_system_output()
+            Generates connections between dataset write and system outputs.
+
+        generate()
+            Generates all the needed connections for data dependency mapping graph.
+    """
+    def __init__(self, dataset_count, dataset_count_map, system_count, system_count_map, dataset_read_count,
+                 system_input_count, dataset_write_count, system_output_count, dataset_read_count_map,
+                 dataset_write_count_map, system_input_count_map, system_output_count_map):
```

Comment:
That is a lot of parameters! Could these be split into multiple groups and moved into separate classes? eg what's dataset related in one DatasetStats class, what's System related in one SystemStats class.